### PR TITLE
refactor(RezervoClass): add bookingOpensAt and bookingClosesAt

### PR DESCRIPTION
**Pull request by AdrianAndersen**
We need this refactoring to so that we can use the actual time the booking opens to schedule the automatic booking

Also, the final commit of this PR enables https://github.com/mathiazom/rezervo/pull/19.

### CODE REVIEW

File: src/lib/integrations/sit/adapters.ts [hunk 1]
```diff
@@ -3,18 +3,24 @@ import { IntegrationIdentifier } from "@/lib/integrations/active";
...
+        startTime: LocalizedDateTime.fromISO(startTime),
+        endTime: LocalizedDateTime.fromISO(convertToProperISO(sitClass.to)),
         location: {
             id: sitClass.studio.id,
             studio: sitClass.studio.name,
             room: sitClass.room,
         },
-        isBookable: sitClass.bookable,
+        bookingOpensAt: LocalizedDateTime.fromISO(convertToProperISO(sitClass.bookingOpensAt)),
+        bookingClosesAt: LocalizedDateTime.fromISO(startTime).minus({ minute: 5 }), // https://www.sit.no/content/trening-slik-booker-du
```
Comment (mathiazom) [inline src/lib/integrations/sit/adapters.ts:23]: Hopefully this is equivalent :crossed_fingers:

**Comment by mathiazom** [comment]
We need some Alembic migration here to support configurations using the old format (i.e. missing the `booking_opens_at`). Should be able to use the old technique of calculating booking time from class start time (at least for Sit, and I don't think we need to support FSC for this).

**Review by mathiazom — APPROVED** [review]

File: src/components/configuration/ConfigBar.tsx [hunk 2]
```diff
@@ -77,6 +77,7 @@ function ConfigBar({
                     weekday: zeroIndexedWeekday(weekday),
                     studio: c.location.id,
                     time: { hour, minute },
+                    booking_opens_at: c.bookingOpensAt.toISO(),
```
Comment (mathiazom) [inline src/components/configuration/ConfigBar.tsx:80]: I think `booking_opens_at` should be left out here, and that the booking scheduling should be performed on the backend. In the future, we might even move all this class config info to the backend and only submit a list of ids.

**Review by mathiazom — CHANGES_REQUESTED** [review]
Should not submit `booking_opens_at` to backend

**Comment by AdrianAndersen** [comment]
We do not need this anymore, since we do not need to pass the integration nor the opening_time

**Pull request closed without merging**